rickynyairo/StackOverflow-lite
# Created endpoints to serve the user with data

#### What does this PR do?
Generates the required endpoints in order to communicate with the API

#### Description of Task to be completed?
The PR has created endpoints for:
-Getting all the questions
-Getting a particular question
-Posting a question
-Posting an answer to a question

#### How should this be manually tested?
Clone the repository [here](https://github.com/rickynyairo/StackOverflow-lite/)

>git clone https://github.com/rickynyairo/StackOverflow-lite/ "new directory"

Navigate to the API directory

>cd API

install the requirements (this should be done in a virtual environment)

> pip install -r requirements.txt 

While in the API (root) folder, access the latest branch:

>git checkout ft-post-an-answer-to-a-question-159749332

Perform tests using nosetests:

>nosetests app/tests -v

#### Any background context you want to provide?
The API defines a set of endpoints for the StackOverflow-Lite project. By creating the above endpoints, the users will be able to communicate with my API
 
#### What are the relevant pivotal tracker stories?
[#159747662](https://www.pivotaltracker.com/story/show/159747662)
[#159748570](https://www.pivotaltracker.com/story/show/159748570)
[#159748250](https://www.pivotaltracker.com/story/show/159748250)
[#159749332](https://www.pivotaltracker.com/story/show/159749332)

#### Questions:
What improvements would you suggest for the functionality?
